### ISSUE TITLE
imagedata.mappixel only maps first line

### ISSUE THREAD

**Issue by Shadowblitz16** [endlesstravel/Love2dCS#89]
I seem to be getting this 
![image](https://user-images.githubusercontent.com/2453707/65847804-e9dec100-e2f7-11e9-8032-e83365b2cc19.png)

from this..
```
    public class Img
    {
        protected Quad       Quad;
        protected Image      Image;
        protected ImageData  ImageData;

        public int Width  => ImageData.GetWidth();
        public int Height => ImageData.GetHeight();
        public int Length => Width * Height;

        public Pal Pal { get; set; } = new Pal();

        public Img(string filename)
        {
            ImageData = Resource.NewImageData(@"Content//"+filename);
            ImageData.MapPixel(MapPixel);
            Quad      = Graphics.NewQuad(0, 0, 8, 8, Width, Height);
        }

        private Vector4 MapPixel(int x, int y, Vector4 p)
        {
            var col = Pal.ClosestColor(new Color(p.r, p.g, p.b, p.a));
            return new Vector4(col.Rf, col.Gf, col.Bf, col.Af);
        }
```

```
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Love;

namespace BBEngine.Library.Gfx
{
    public class Pal
    {
        List<Color> Colors = new List<Color>();

        public Color this[int i]
        {
            get
            {
                return Colors[i];
            }
            set
            {
                Colors[i] = value;
            }
        }
        public int Length => Colors.Count;

        public Pal(string filename)
        {
            var Data  = Resource.NewImageData(@"Content//" + filename);
            var DataW = Data.GetWidth ();
            var DataH = Data.GetHeight();
            var DataL = DataW * DataH;
            for (var i=0; i< DataL; i++)
            {
                var pix = Data.GetPixel((int)i % DataW, (int)i / DataW);
                Colors.Add(new Color(pix.intValue0, pix.intValue1, pix.intValue2, pix.intValue3));
            }
        }

        public Pal()
        {
            var Data = Resource.NewImageData("Content\\pal.png");
            var DataW = Data.GetWidth();
            var DataH = Data.GetHeight();
            var DataL = DataW * DataH;
            for (var i = 0; i < DataL; i++)
            {
                var pix = Data.GetPixel((int)i % DataW, (int)i / DataW);
                Colors.Add(new Color((byte)pix.rgba8.r, (byte)pix.rgba8.g, (byte)pix.rgba8.b, (byte)pix.rgba8.a));
            }
        }


        private int ColorDiff(Color c1, Color c2)
        {
            return (int)Math.Sqrt
            (
                (c1.r - c2.r) * (c1.r - c2.r) +
                (c1.g - c2.g) * (c1.g - c2.g) +
                (c1.b - c2.b) * (c1.b - c2.b)
            );
        }
        public Color ClosestColor(Color target)
        {
            return Colors[ClosestIndex(target)];
        }
        public int ClosestIndex(Color target)
        {
            var colorDiffs = Colors.Select(n => ColorDiff(n, target)).Min(n => n);
            return Colors.FindIndex(n => ColorDiff(n, target) == colorDiffs);
        }
    }
}
```

note that only the first line and half the second are being mapped

**Comment by endlesstravel** [comment]
Thanks report,  there's a problem on `ImageData:MapPixel`. I'll fix it.
`ImageData:SetPixelsFloat` can be used as a temporary alternative

**Comment by endlesstravel** [comment]
The bugs on `ImageData:MapPixel` has fixed on version of  11.0.39 
It should be ok now.

https://www.nuget.org/packages/Love2dCS/11.0.39

**Comment by endlesstravel** [comment]
closed because it fixed.
